### PR TITLE
[FLINK-11362] Port TaskManagerComponentsStartupShutdownTest to new co…

### PR DESCRIPTION
**Pull request by tisonkun**
## What is the purpose of the change

*Port `TaskManagerComponentsStartupShutdownTest` to `TaskExecutorTest#testTaskManagerServicesShutdown`.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector:(no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

cc @tillrohrmann

### CODE REVIEW

**Comment by rmetzger** [comment]
@GJL I saw you've ticked an item in the review list from the bot.
How it actually works is like this: 
@flinkbot approve description

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java [hunk 1]
```diff
@@ -230,6 +233,78 @@ public void teardown() throws Exception {
...
+	@Test
+	public void testTaskManagerServicesShutdown() throws Exception {
+		final TaskSlotTable taskSlotTable = new TaskSlotTable(Collections.singleton(ResourceProfile.UNKNOWN), timerService);
+
+		final JobLeaderService jobLeaderService = new JobLeaderService(taskManagerLocation);
+
+		final long heartbeatInterval = 1000L;
+		final long heartbeatTimeout = 1000L;
+
+		final HeartbeatServices heartbeatServices = new HeartbeatServices(heartbeatInterval, heartbeatTimeout);
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java]: Definition/declaration should be closer to `TaskExecutor` creation.

**Comment by GJL** [comment]
Thank you for your contribution to Apache Flink @TisonKun

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java [hunk 2]
```diff
@@ -230,6 +233,78 @@ public void teardown() throws Exception {
...
+		final IOManager ioManager = new IOManagerAsync(new String[]{ tmp.newFolder().getAbsolutePath() });
+
+		final TaskExecutorLocalStateStoresManager localStateStoresManager = new TaskExecutorLocalStateStoresManager(
+			false,
+			ioManager.getSpillingDirectories(),
+			Executors.directExecutor());
+
+		final int networkBufNum = 32;
+		final int bufferSize = 32 * 1024;
+		final NetworkEnvironmentConfiguration netConf = new NetworkEnvironmentConfiguration(
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java]: `netConf` is not really needed

**Comment by GJL** [comment]
@flinkbot disapprove architecture

**Comment by GJL** [comment]
@flinkbot approve architecture

**Comment by GJL** [comment]
@flinkbot disapprove description

**Comment by GJL** [comment]
@flinkbot approve description

**Comment by GJL** [comment]
@flinkbot approve all

**Comment by rmetzger** [comment]
The bot has still not updated the tracking comment. I'm looking into it.

**Comment by GJL** [comment]
@flinkbot approve all

**Comment by GJL** [comment]
Ugh... 🤦‍♂️

**Comment by GJL** [comment]
@rmetzger Let me know if you find something.

**Comment by rmetzger** [comment]
Yes, I will. I was able to reproduce the issue. Re-sending the message will not fix the problem. Mentioning the bot again might help.
I'll let you know when there's an update

**Comment by TisonKun** [comment]
Thanks for your review and improvement @GJL

**Pull request closed without merging**